decidim-core/lib/decidim/coauthorable.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This concern contains the logic related to collective or shared authorship.
+  #
+  # Coauthorable shares nearly the same object interface as Authorable but with some differences.
+  # - `authored_by?(user)` is exactly the same
+  # - `normalized_authors` is pluralized
+  #
+  # All coauthors, including the initial author, will share the same permissions.
+  module Coauthorable
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :coauthorships, as: :coauthorable, class_name: "Decidim::Coauthorship", dependent: :destroy
+      has_many :authors, through: :coauthorships, class_name: "Decidim::User"
+      has_many :user_groups, as: :coauthorable, class_name: "Decidim::UserGroup", through: :coauthorships
+
+      scope :from_author, ->(author) { joins(:coauthorships).where("decidim_coauthorships.decidim_author_id": author.id) }
```

Comment:
We should probably check that the `decidim_user_group_id` column is null

Reply:
It is correct this way. But I'll add some documentation to make it clear that it retrieves models from all identities of the user.

Reply:
Now I see in ProposalsController that it also has the behaviour you defined so I'll add a new method to disambiguate.

Reply:
Nice, thanks!